src/main/java/com/example/bookstore/model/CartItem.java

```diff
@@ -0,0 +1,34 @@
+package com.example.bookstore.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Table(name = "cart_items")
+@Getter
+@Setter
+public class CartItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "shopping_cart_id", referencedColumnName = "id", nullable = false)
```

Comment:
just wondering

`@JoinColumn(nullable = false)` == `@AnyToOne(optional = false)` ? :)

 here you specified just `nullable`, but do you need to specify `optional` as well?

try to find out whether these both are equal

Reply:
If i correctly understood it's not equal. nullable sets notnull constraint in DB. And optional just checks it in runtime.  Those are both use for null-check, but optional do it much faster

Reply:
@teract10s good, then you definitely need optional

and if your ddl-auto is `validate` then nullable too :)